module/pattern_mode.c

```diff
@@ -168,6 +202,124 @@ void process_pattern_keys(uint8_t k, uint8_t m, bool is_held_key) {
             dirty = true;
         }
     }
+    // alt-[: decrement by 1 semitone
+    else if (match_alt(m, k, HID_OPEN_BRACKET)) {
+        if (editing_number) {
+            edit_buffer = note_decrement(edit_buffer, 1);
+            dirty = true;
+        }
+        else {
+            int16_t pattern_val =
+                ss_get_pattern_val(&scene_state, pattern, base + offset);
+            int16_t new_val = note_decrement(pattern_val, 1);
+            ss_set_pattern_val(&scene_state, pattern, base + offset, new_val);
+            dirty = true;
+        }
```

Comment:
this block repeats below with the only difference being the inc/dec value - extract in a function? (and it could take a signed int for parameter so you could use the same function for inc and dec)

Reply:
https://github.com/monome/teletype/pull/194/files#diff-d7723468801034a237015ca6ef3b6479R111-R134